### PR TITLE
Document unshelve (#3225)

### PR DESCRIPTION
**Pull request by Blaisorblade**
I believe this is the appropriate place for users to read about this, even tho `unshelve` is maybe technically a tactical (but then, is `intuition` a tactical as well?).

This example was explicitly requested in #3225 and is commonly used in both
Iris (and was documented in the changelog at the time).

**Fixes:** #4508.

**Kind:** documentation.

### CODE REVIEW

**Comment by ejgallego** [comment]
cc: #6292 #8770

**Review by Zimmi48 — APPROVED** [review]
Thanks! The location seems right indeed. Tactic or tactical: it doesn't matter much. It seems to me that `unshelve` is more of a tactic than a Ltac construct. I'll wait until later today to merge to give time to @ppedrot to object if I'm wrong.

**Comment by ppedrot** [comment]
Tactical is just Ltac jargon to describe Ltac terms of moral type `(unit -> unit) -> unit` when tactics have type `unit -> unit`. So, it doesn't matter much.

**Comment by Zimmi48** [comment]
@Blaisorblade Your use of formatting and colon around `Fixes` broke GitHub's automation (auto-closing the corresponding issue).

**Comment by Blaisorblade** [comment]
@Zimmi48 silly me, thanks for the fix :-|

**Pull request merged**
